flyio/caveats.go

```diff
@@ -314,3 +315,53 @@ func (c *NoAdminFeatures) Prohibits(a macaroon.Access) error {
 
 	return nil
 }
+
+// Commands is a list of commands allowed by this token.
+// The zero value rejects any command.
+type Commands []Command
+
+// Command is a single command to allow. The zero value allows any command.
+// If exact is true, the args must match exactly. Otherwise the args must
+// match the prefix of the command being executed.
+type Command struct {
+	Args  []string `json:"args"`
+	Exact bool     `json:"exact,omitempty"`
+}
+
+func init()                                         { macaroon.RegisterCaveatType(&Commands{}) }
+func (c *Commands) CaveatType() macaroon.CaveatType { return CavCommands }
+func (c *Commands) Name() string                    { return "Commands" }
+
+func (c *Commands) Prohibits(a macaroon.Access) error {
+	f, isFlyioAccess := a.(CommandGetter)
+	if !isFlyioAccess {
+		return fmt.Errorf("%w: access isnt CommandGetter", macaroon.ErrInvalidAccess)
+	}
+
+	commandArgs := f.GetCommand()
+	if commandArgs == nil {
+		return fmt.Errorf("%w: only authorized for command execution", resset.ErrResourceUnspecified)
+	}
+
+	var found bool
+	allowedCommands := *c
+	for _, allowedCommand := range allowedCommands {
+		if len(allowedCommand.Args) > len(commandArgs) {
+			continue
+		}
+
+		if allowedCommand.Exact && len(allowedCommand.Args) != len(commandArgs) {
+			continue
+		}
+
+		if !slices.Equal(allowedCommand.Args, commandArgs[:len(allowedCommand.Args)]) {
+			continue
+		}
+		found = true
```

Comment:
this follows the pattern in the Mutations access check, and I think it might be clearer having the success returned at the end of the function.  I'll add a break after found, though the savings will be miniscule.